src/main/java/com/es/phoneshop/web/ProductDetailsPageServlet.java

```diff
@@ -0,0 +1,27 @@
+package com.es.phoneshop.web;
+
+import com.es.phoneshop.model.product.ArrayListProductDao;
+import com.es.phoneshop.model.product.ProductDao;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+public class ProductDetailsPageServlet extends HttpServlet {
+
+    ProductDao productDao;
+
+    @Override
+    public void init() {
+        productDao = ArrayListProductDao.getInstance();
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException{
+        String productId = request.getPathInfo();
+        request.setAttribute("product", productDao.getProduct(Long.valueOf(productId.substring(1))));
```

Comment:
Do not believe user input: what if the user passes a string instead of a number in the request?